controllers/awscluster_controller.go

```diff
@@ -98,6 +99,12 @@ func (r *AWSClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reter
 
 	// Always close the scope when exiting this function so we can persist any AWSCluster changes.
 	defer func() {
+		if c := conditions.Get(clusterScope.AWSCluster, infrav1.BastionHostReadyCondition); c != nil {
```

Comment:
Would it be better to use the spec configuration related to whether a bastion instance should be configured for this rather than trying to determine from the conditions here?

Reply:
yep, agree with not checking a condition that the controller itself sets. I avoided it since there were multiple reasons for it to silently skip bastion creation:
- Unmanaged VPC (updating it to check for that)
- Bastion disabled in spec (updating it to check for that)
- Missing private subnet. (This one feels like it's expected to get resolved in future reconciliations so not including this in the check)